### PR TITLE
Final step in converting the data readers

### PR DESCRIPTION
**Pull request by jeremykubica**
Replaces the code path in sorcha.py with the a new `CombinedDataReader` that will do chunked reading of all the input files (ephemeris and auxiliary files). This class provides the ability to read in chunks based on any of the data files, including chunks of the ephemeris file.

Adapts the current test_PPReadAllInput.py to cover the behavior of the new code path.

- [ ] Does pip install still work?
- [X] Have you written a unit test for any new functions?
- [X] Do all the units tests run successfully?
- [X] Does Sorcha run successfully on a test set of input files/databases?
- [X] Have you used black on the files you have updated to confirm python programming style guide enforcement?

### CODE REVIEW

File: src/sorcha/readers/CombinedDataReader.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        # No more data to load.
+        if len(obj_ids) == 0:
+            return None
+
+        # Load in the data for this block.
+        if ephem_df is None:
+            ephem_df = self.ephem_reader.read_objects(obj_ids)
+        ephem_ids = set(ephem_df["ObjID"].unique().tolist())
+
+        for reader in self.aux_data_readers:
```
Comment (drewoldag) [inline src/sorcha/readers/CombinedDataReader.py:109]: This for loop isn't actually doing another full read of the first aux_reader, right? I think that it's using the inverse index to read in only the required rows, but I wanted to make sure that my understanding is correct.

File: src/sorcha/sorcha.py [hunk 2]
```diff
@@ -89,11 +94,28 @@ def runLSSTPostProcessing(cmd_args):
...
+        ephem_type = configs["ephemerides_type"]
+        if ephem_type.casefold() != "oif":  # pragma: no cover
+            pplogger.error(f"PPReadAllInput: Unsupported value for ephemerides_type {ephem_type}")
+            sys.exit(f"PPReadAllInput: Unsupported value for ephemerides_type {ephem_type}")
+        reader.add_ephem_reader(OIFDataReader(cmd_args["oifoutput"], configs["eph_format"]))
+
+    reader.add_aux_data_reader(OrbitAuxReader(cmd_args["orbinfile"], configs["aux_format"]))
+    reader.add_aux_data_reader(CSVDataReader(cmd_args["paramsinput"], configs["aux_format"]))
+    if configs["comet_activity"] == "comet":
+        reader.add_aux_data_reader(CSVDataReader(cmd_args["cometinput"], configs["aux_format"]))
```
Comment (drewoldag) [inline src/sorcha/sorcha.py:115]: This looks so good. One thing that has come up with the light curve variability work is that there's a reasonably good chance that users will create new functions that require parameters beyond what is currently read in. 

This works sets us up so nicely for that.

**Review by drewoldag — APPROVED** [review]
This is awesome work. Thanks for putting it together!

File: src/sorcha/readers/CombinedDataReader.py (hunk 1 above)
Comment (jeremykubica) [reply src/sorcha/readers/CombinedDataReader.py:109]: That is correct. Outside the for loop, the program is reading in a set range of rows from `aux_data_readers[0]`. During the first iteration of the for loop, the program will re-open the file and load the rows for the matching object IDs.

That being said, this is still reopening the file and reading it back in. Let me update it to only do that if it has not already loaded the `aux_data_readers[0]` data.

**Review by drewoldag — APPROVED** [review]
That change looks good to me.

**Pull request merged**